### PR TITLE
release v1.0.2

### PR DESCRIPTION
**Pull request by weekface**
### Check List 

Tests 

Code changes

 - Has Helm charts change
 - Has documents change

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

**Comment by weekface** [comment]
@yikeke PTAL

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,57 @@
...
+
+But TiKV pod uses instance store to store its data. When the instance is replaced, all the data on the instance store will be lost. TiKV has to resync all data to the newly added instance. Though TiDB is a distributed database and can work when a node fails, the cost to resync data is quite big if the dataset is large. Besides, the ec2 instance may recover to a healthy status by rebooting.
+
+So we disable the auto-scaling-group's replace behavior in `v1.0.2`.
+
+Auto-scaling-group scaling process can also be suspended according to its [documentation](https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-suspend-resume-processes.html) if you are using `v1.0.1` or prior versions.
+
+### Improvements
+
+- Migrating statefulsets apiVersion to `app/v1`
```
Comment (tennix) [inline CHANGELOG.md]: This is a bugfix to fix compatiblity with k8s 1.16 and above versions.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,57 @@
...
+- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in aws/gcp/aliyun
+- Add provider and module versions for aws
+
+### Bug fixes
+
+- Fix that the create_tidb_cluster_release variable in AWS terraform script does not work
+- Fix compatibility issues for ([#1012](https://github.com/pingcap/tidb-operator/pull/1012)): add `v1beta1` to statefulset apiVersions
+- Fix the issue that TiDB Loadbalancer is empty in terraform output
+- Fix a compatibility issue on TiKV maxFailoverCount
+- Fix provider versions constraint issues for GCP and Aliyun
```
Comment (tennix) [inline CHANGELOG.md]:
```suggestion
- Fix Terraform providers version constraint issues for GCP and Aliyun
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,57 @@
...
+- Fix values file customization for tidb-operator on Aliyun
+- Fix tidb-operator crash when user modify sts upgrade strategy improperly
```
Comment (tennix) [inline CHANGELOG.md]:
```suggestion
- Fix tidb-operator crash when users modify statefulset upgrade strategy improperly
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,57 @@
...
+- Fix drainer misconfiguration
+
+## Detailed Bug Fixes and Changes
+
+- Fix that the create_tidb_cluster_release variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```
Comment (tennix) [inline CHANGELOG.md]:
```suggestion
- Fix that the `create_tidb_cluster_release` variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,57 @@
...
+- Migrating statefulsets apiVersion to `app/v1`
+- Suspend ReplaceUnhealthy process for AWS tikv auto-scaling-group
+- Add a new VM manager `qm` in stability test
+- Add `tikv.maxFailoverCount` limit to TiKV
+- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in aws/gcp/aliyun
+- Add provider and module versions for aws
+
+### Bug fixes
+
+- Fix that the create_tidb_cluster_release variable in AWS terraform script does not work
```
Comment (tennix) [inline CHANGELOG.md]:
```suggestion
- Fix that the `create_tidb_cluster_release` variable in AWS terraform script does not work
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,57 @@
+# TiDB Operator v1.0.2 Release Notes
+
+## v1.0.2 What's New
+
+### Action Required
+
+The AWS Terraform script uses auto-scaling-group for all components (pd/tikv/tidb/monitor), when an ec2 instance fails the health check, the ec2 instance will be replaced. This is helpful for stateless applications or applications using EBS volumes to store data.
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
The AWS Terraform script uses auto-scaling-group for all components (PD/TiKV/TiDB/monitor). When an ec2 instance fails the health check, the instance will be replaced. This is helpful for those applications that are stateless or use EBS volumes to store data.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,57 @@
...
+
+But TiKV pod uses instance store to store its data. When the instance is replaced, all the data on the instance store will be lost. TiKV has to resync all data to the newly added instance. Though TiDB is a distributed database and can work when a node fails, the cost to resync data is quite big if the dataset is large. Besides, the ec2 instance may recover to a healthy status by rebooting.
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
But a TiKV Pod uses instance store to store its data. When an instance is replaced, all the data on its store will be lost. TiKV has to resync all data to the newly added instance. Though TiDB is a distributed database and can work when a node fails, resyncing data can cost much if the dataset is large. Besides, the ec2 instance may be recovered to a healthy state by rebooting.
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,57 @@
...
+
+So we disable the auto-scaling-group's replace behavior in `v1.0.2`.
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
So we disabled the auto-scaling-group's replacing behavior in `v1.0.2`.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,57 @@
...
+
+Auto-scaling-group scaling process can also be suspended according to its [documentation](https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-suspend-resume-processes.html) if you are using `v1.0.1` or prior versions.
+
+### Improvements
+
+- Suspend ReplaceUnhealthy process for AWS tikv auto-scaling-group
+- Add a new VM manager `qm` in stability test
+- Add `tikv.maxFailoverCount` limit to TiKV
+- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in aws/gcp/aliyun
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in AWS/GCP/Aliyun
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,57 @@
...
+
+But TiKV pod uses instance store to store its data. When the instance is replaced, all the data on the instance store will be lost. TiKV has to resync all data to the newly added instance. Though TiDB is a distributed database and can work when a node fails, the cost to resync data is quite big if the dataset is large. Besides, the ec2 instance may recover to a healthy status by rebooting.
+
+So we disable the auto-scaling-group's replace behavior in `v1.0.2`.
+
+Auto-scaling-group scaling process can also be suspended according to its [documentation](https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-suspend-resume-processes.html) if you are using `v1.0.1` or prior versions.
+
+### Improvements
+
+- Suspend ReplaceUnhealthy process for AWS tikv auto-scaling-group
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Suspend ReplaceUnhealthy process for AWS TiKV auto-scaling-group
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,57 @@
...
+- Add a new VM manager `qm` in stability test
+- Add `tikv.maxFailoverCount` limit to TiKV
+- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in aws/gcp/aliyun
+- Add provider and module versions for aws
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Add provider and module versions for AWS
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,57 @@
...
+
+### Bug fixes
+
+- Migrating statefulsets apiVersion to `app/v1` which fixes compatiblity with k8s 1.16 and above versions
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Migrate statefulsets apiVersion to `app/v1` which fixes compatibility with Kubernetes 1.16 and above versions
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,57 @@
...
+- Fix that the `create_tidb_cluster_release` variable in AWS terraform script does not work
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix the issue that the `create_tidb_cluster_release` variable in AWS Terraform script does not work
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,57 @@
...
+- Fix compatibility issues for ([#1012](https://github.com/pingcap/tidb-operator/pull/1012)): add `v1beta1` to statefulset apiVersions
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix compatibility issues by adding `v1beta1` to statefulset apiVersions
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,57 @@
...
+- Fix the issue that TiDB Loadbalancer is empty in terraform output
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix the issue that TiDB Loadbalancer is empty in Terraform output
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,57 @@
...
+- Fix a compatibility issue on TiKV maxFailoverCount
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix a compatibility issue of TiKV `maxFailoverCount`
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,57 @@
...
+- Fix Terraform providers version constraint issues for GCP and Aliyun
+- Fix values file customization for tidb-operator on Aliyun
+- Fix tidb-operator crash when users modify statefulset upgrade strategy improperly
+- Fix drainer misconfiguration
+
+## Detailed Bug Fixes and Changes
+
+- Fix that the create_tidb_cluster_release variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix the issue that the `create_tidb_cluster_release` variable in AWS Terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,3 +1,57 @@
...
+- Fix compatibility issues for ([#1012](https://github.com/pingcap/tidb-operator/pull/1012)): add `v1beta1` to statefulset apiVersions ([#1054](https://github.com/pingcap/tidb-operator/pull/1054))
+- Enable ConfigMapRollout by default in stability test ([#1036](https://github.com/pingcap/tidb-operator/pull/1036))
+- Fix the issue that TiDB Loadbalancer is empty in terraform output ([#1045](https://github.com/pingcap/tidb-operator/pull/1045))
+- Migrating statefulsets apiVersion to `app/v1` which fixes compatiblity with k8s 1.16 and above versions ([#1012](https://github.com/pingcap/tidb-operator/pull/1012))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Migrate statefulsets apiVersion to `app/v1` which fixes compatibility with Kubernetes 1.16 and above versions ([#1012](https://github.com/pingcap/tidb-operator/pull/1012))
```

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,3 +1,57 @@
...
+- Fix values file customization for tidb-operator on Aliyun
+- Fix tidb-operator crash when users modify statefulset upgrade strategy improperly
+- Fix drainer misconfiguration
+
+## Detailed Bug Fixes and Changes
+
+- Fix that the create_tidb_cluster_release variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
+- Fix compatibility issues for ([#1012](https://github.com/pingcap/tidb-operator/pull/1012)): add `v1beta1` to statefulset apiVersions ([#1054](https://github.com/pingcap/tidb-operator/pull/1054))
+- Enable ConfigMapRollout by default in stability test ([#1036](https://github.com/pingcap/tidb-operator/pull/1036))
+- Fix the issue that TiDB Loadbalancer is empty in terraform output ([#1045](https://github.com/pingcap/tidb-operator/pull/1045))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix the issue that TiDB Loadbalancer is empty in Terraform output ([#1045](https://github.com/pingcap/tidb-operator/pull/1045))
```

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,3 +1,57 @@
...
+- Migrating statefulsets apiVersion to `app/v1` which fixes compatiblity with k8s 1.16 and above versions ([#1012](https://github.com/pingcap/tidb-operator/pull/1012))
+- Only expect upgrade complete when rollback bad configuration in stability test ([#1030](https://github.com/pingcap/tidb-operator/pull/1030))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Only expect upgrade complete when rollback bad configuration in stability test ([#1030](https://github.com/pingcap/tidb-operator/pull/1030))
```
Please explain this sentence in Chinese. It is confusing me.

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,3 +1,57 @@
...
+- Suspend ReplaceUnhealthy process for AWS tikv auto-scaling-group ([#1014](https://github.com/pingcap/tidb-operator/pull/1014))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Suspend ReplaceUnhealthy process for AWS TiKV auto-scaling-group ([#1014](https://github.com/pingcap/tidb-operator/pull/1014))
```

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,3 +1,57 @@
...
+- Add a new VM manager `qm` in stability test ([#896](https://github.com/pingcap/tidb-operator/pull/896))
+- Fix provider versions constraint issues for GCP and Aliyun ([#959](https://github.com/pingcap/tidb-operator/pull/959))
+- Fix values file customization for tidb-operator on aliyun ([#971](https://github.com/pingcap/tidb-operator/pull/971))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix values file customization for tidb-operator on Aliyun ([#971](https://github.com/pingcap/tidb-operator/pull/971))
```

File: CHANGELOG.md [hunk 23]
```diff
@@ -1,3 +1,57 @@
...
+- Fix a compatibility issue on TiKV `tikv.maxFailoverCount` ([#977](https://github.com/pingcap/tidb-operator/pull/977))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix a compatibility issue of TiKV `tikv.maxFailoverCount` ([#977](https://github.com/pingcap/tidb-operator/pull/977))
```

File: CHANGELOG.md [hunk 24]
```diff
@@ -1,3 +1,57 @@
...
+- Add `tikv.maxFailoverCount` limit to TiKV ([#965](https://github.com/pingcap/tidb-operator/pull/965))
+- Fix tidb-operator crash when user modify sts upgrade strategy improperly ([#912](https://github.com/pingcap/tidb-operator/pull/912))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix tidb-operator crash when users modify statefulset upgrade strategy improperly ([#912](https://github.com/pingcap/tidb-operator/pull/912))
```

File: CHANGELOG.md [hunk 25]
```diff
@@ -1,3 +1,57 @@
...
+- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in aws/gcp/aliyun ([#947](https://github.com/pingcap/tidb-operator/pull/947))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in AWS/GCP/Aliyun ([#947](https://github.com/pingcap/tidb-operator/pull/947))
```

File: CHANGELOG.md [hunk 26]
```diff
@@ -1,3 +1,57 @@
...
+- Add note about setting pv reclaim policy to retain ([#911](https://github.com/pingcap/tidb-operator/pull/911))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Add note about setting PV reclaim policy to retain ([#911](https://github.com/pingcap/tidb-operator/pull/911))
```

File: CHANGELOG.md [hunk 27]
```diff
@@ -1,3 +1,57 @@
...
+- Fix drainer misconfiguration ([#939](https://github.com/pingcap/tidb-operator/pull/939))
+- Add provider and module versions for aws ([#926](https://github.com/pingcap/tidb-operator/pull/926))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Add provider and module versions for AWS ([#926](https://github.com/pingcap/tidb-operator/pull/926))
```

File: CHANGELOG.md (hunk 20 above)
Comment (weekface) [reply CHANGELOG.md]: @aylei PTAL

File: CHANGELOG.md [hunk 28]
```diff
@@ -1,3 +1,59 @@
...
+
+- Suspend ReplaceUnhealthy process for AWS TiKV auto-scaling-group
+- Add a new VM manager `qm` in stability test
+- Add `tikv.maxFailoverCount` limit to TiKV
+- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in AWS/GCP/Aliyun
+- Add provider and module versions for AWS
+
+### Bug fixes
+
+- Fix tkctl version does not work when the release name is un-wanted
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix the issue that tkctl version does not work when the release name is un-wanted
```

File: CHANGELOG.md [hunk 29]
```diff
@@ -1,3 +1,59 @@
...
+- Fix the issue that TiDB Loadbalancer is empty in Terraform output
+- Fix a compatibility issue of TiKV `maxFailoverCount`
+- Fix Terraform providers version constraint issues for GCP and Aliyun
+- Fix values file customization for tidb-operator on Aliyun
+- Fix tidb-operator crash when users modify statefulset upgrade strategy improperly
+- Fix drainer misconfiguration
+
+## Detailed Bug Fixes and Changes
+
+- Fix tkctl version does not work when the release name is un-wanted ([#1065](https://github.com/pingcap/tidb-operator/pull/1065))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix the issue that tkctl version does not work when the release name is un-wanted ([#1065](https://github.com/pingcap/tidb-operator/pull/1065))
```

File: CHANGELOG.md [hunk 30]
```diff
@@ -1,3 +1,59 @@
...
+- Fix that the `create_tidb_cluster_release` variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```
Comment (yikeke) [inline CHANGELOG.md]:
```suggestion
- Fix the issue that the `create_tidb_cluster_release` variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```

File: CHANGELOG.md (hunk 20 above)
Comment (aylei) [reply CHANGELOG.md]: ”在 e2e 测试中回滚错误配置时，（测试用例）只期待（TiDB集群）升级完成“

Comment (yikeke) [reply CHANGELOG.md]: Thanks~ I edited my suggested change accordingly, PTAL @weekface

**Pull request merged**